tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -432,9 +434,35 @@ def data_route(self, request):
       return http_util.Respond(request, '404 Not Found', 'text/plain', code=404)
     return http_util.Respond(request, data, 'application/json')
 
+  @wrappers.Request.application
+  def capture_route(self, request):
+    service_addr = request.args.get('service_addr')
+    try:
+      duration = int(request.args.get('duration'))
+    except TypeError:
+      return http_util.Respond(request, 'Invalid duration', 
+                               'text/plain', code=400)
+    is_tpu_name = request.args.get('is_tpu_name') == 'true'
+    if is_tpu_name:
+      tpu_cluster_resolver = (
+          tf.distribute.cluster_resolver.TPUClusterResolver([service_addr]))
+      service_addr = tpu_cluster_resolver.get_master()
+      # TPU cluster resolver always returns port 8470. Replace it with 8466
+      # on which profiler service is running.
+      service_addr = service_addr.replace('grpc://', '').replace(':8470', ':8466')
+    try:
+      profiler_client.start_tracing(service_addr, self.logdir, duration)
```

Comment:
Oh interesting, didn't realize there is no way to stop and have to wait for that duration. Just curious, why?

Reply:
Separately, I am pondering the implication of using `self.logdir` here. Are profile data tagged with timestamp or anything? Would one be able to make any sense of the name when user profiles multiple times?

Reply:
1) I think it is fine to block the thread since python server uses different multiple threads to handle request.
2) Profile data are tagged with timestamp.